packages/headless/src/features/folding/folding-actions.ts

```diff
@@ -95,3 +97,18 @@ export const loadCollection = createAsyncThunk<
     return {collectionId, results: response.success.results};
   }
 );
+
+function getQForHighlighting(state: StateNeededByLoadCollection) {
+  // This piece of code serves the following purpose:
+  // Inject the "original" query "q" to get proper keywords highlighting when loading a full collection
+  // However, the intent behind this feature is to load "every results available for this collection", regardless of other end user filters (including the search box itself)
+  // For that reason, we force enable query syntax + inject an `OR @uri` expression in the query.
+
+  if (state.query.q === '') {
+    return '';
+  }
+
+  return state.query.enableQuerySyntax
+    ? `${state.query.q} OR @uri`
+    : `( <@- ${state.query.q} -@> ) OR @uri`;
```

Comment:
Are there any tests for this logic? I find test descriptions are a good place to add explanations on why something is a certain way. e.g.

```
it("when query syntax is active, it adds @uri to guarantee all results containing the query are loaded", ...)
```

Reply:
Yeah I can add more tests around the query creation part itself.